### PR TITLE
[HOTFIX] Enhanced compatibility with Spark's FilePartition interface

### PR DESCRIPTION
**Pull request by marchpure**
### Why is this PR needed?
 Input type of Spark's FilePartition will change from Seq to Array.
We shall create filepartition by entering ARRAY type, which is compatible with both Array and Seq.
 
 ### What changes were proposed in this PR?
create filepartition by entering ARRAY type, instead of entering Seq type.
    
 ### Does this PR introduce any user interface change?
 - No

 ### Is any new testcase added?
 - No

### CODE REVIEW

**Comment by CarbonDataQA1** [comment]
Build Failed  with Spark 2.3.4, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.3/2441/

**Comment by CarbonDataQA1** [comment]
Build Failed  with Spark 2.3.4, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.3/2444/

**Comment by CarbonDataQA1** [comment]
Build Success with Spark 2.4.4, Please check CI http://121.244.95.60:12545/job/ApacheCarbon_PR_Builder_2.4.4/736/

**Comment by CarbonDataQA1** [comment]
Build Success with Spark 2.3.4, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.3/2446/

**Comment by CarbonDataQA1** [comment]
Build Success with Spark 2.4.4, Please check CI http://121.244.95.60:12545/job/ApacheCarbon_PR_Builder_2.4.4/738/

File: integration/spark/src/main/spark2.3/org/apache/carbondata/spark/adapter/CarbonToSparkAdapter.scala [hunk 1]
```diff
@@ -17,12 +17,12 @@
...
 package org.apache.carbondata.spark.adapter
 
-import scala.collection.mutable.ArrayBuffer
+import scala.collection.mutable.{ArrayBuffer, WrappedArray}
 
 import org.apache.spark.sql.execution.datasources.{FilePartition, PartitionedFile}
 
 object CarbonToSparkAdapter {
   def createFilePartition(index: Int, files: ArrayBuffer[PartitionedFile]) = {
-    FilePartition(index, files.toArray.toSeq)
```
Comment (QiangCai) [inline integration/spark/src/main/spark2.3/org/apache/carbondata/spark/adapter/CarbonToSparkAdapter.scala]: better to modify CsvRDDHelper directly and remove adapter/CarbonToSparkAdapter.scala （both 2.3 and 2.4）

**Comment by CarbonDataQA1** [comment]
Build Failed  with Spark 2.4.4, Please check CI http://121.244.95.60:12545/job/ApacheCarbon_PR_Builder_2.4.4/757/

**Comment by CarbonDataQA1** [comment]
Build Success with Spark 2.3.4, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.3/2465/

**Comment by CarbonDataQA1** [comment]
Build Failed  with Spark 2.4.4, Please check CI http://121.244.95.60:12545/job/ApacheCarbon_PR_Builder_2.4.4/761/

**Comment by CarbonDataQA1** [comment]
Build Failed  with Spark 2.3.4, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.3/2469/

**Comment by CarbonDataQA1** [comment]
Build Failed  with Spark 2.4.4, Please check CI http://121.244.95.60:12545/job/ApacheCarbon_PR_Builder_2.4.4/762/

**Comment by CarbonDataQA1** [comment]
Build Failed  with Spark 2.3.4, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.3/2470/

File: integration/spark/src/main/scala/org/apache/carbondata/spark/load/CsvRDDHelper.scala [hunk 2]
```diff
@@ -94,7 +93,7 @@ object CsvRDDHelper {
     def closePartition(): Unit = {
       if (currentFiles.nonEmpty) {
         val newPartition =
-          CarbonToSparkAdapter.createFilePartition(
+          FilePartition(
```
Comment (Indhumathi27) [inline integration/spark/src/main/scala/org/apache/carbondata/spark/load/CsvRDDHelper.scala]: Should be FilePartition( partitions.size, currentFiles.toArray)?

File: integration/spark/src/main/scala/org/apache/carbondata/spark/load/CsvRDDHelper.scala [hunk 3]
```diff
@@ -20,7 +20,7 @@ package org.apache.carbondata.spark.load
 import java.text.SimpleDateFormat
 import java.util.{Date, Locale}
 
-import scala.collection.mutable.ArrayBuffer
+import scala.collection.mutable.{ArrayBuffer, WrappedArray}
```
Comment (Indhumathi27) [inline integration/spark/src/main/scala/org/apache/carbondata/spark/load/CsvRDDHelper.scala]: Revert this change

**Comment by ajantha-bhat** [comment]
@marchpure : please reply to the comments and get it merged.
